### PR TITLE
Adds no-cache headers for browser

### PR DESCRIPTION
**Pull request by gfiorav**
Please CR, @juanignaciosl 

This only affects client side cache and only our editor pages (which have sensible info).

This will prevent those pages showing up when clicking “back” button after logout in modern browsers.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
I think we should summon @zenitraM for this CR, since it's cache-related.

**Comment by zenitraM** [comment]
Which endpoints does this change? I think only viz.jsons, embeds and public pages go through Varnish, which would be the only one affected by this.

**Comment by gfiorav** [comment]
@zenitraM this affects all pages related to the editor (`/admin`) so https://github.com/CartoDB/cartodb/blob/master/config/routes.rb#L48

I think we're in the clear no?

**Comment by juanignaciosl** [comment]
I was almost sure that we don't cache private pages, but I wanted your confirmation :-)

IMHO, check it in staging and go :+1:

**Comment by zenitraM** [comment]
Public maps and embed maps are under admin:
https://github.com/CartoDB/cartodb/blob/master/config/routes.rb#L283-L284

I think that the rules coming from the server will be discarded by Varnish and will cache the pages anyway - but please test on staging.

You can see if a request is cached by the headers:
- X-Varnish: should have one number if not cached, two numbers if cached
- X-Cache: `HIT` means it was returned by cache

So ensure that:
- requesting a visualization two times results on a `MISS` the first time, `HIT` the second
- altering something on that visualization (zooms, etc) causes then a `MISS` and then a `HIT`

**Comment by gfiorav** [comment]
@zenitraM golden!

Thanks a lot, checking that now.

**Comment by juanignaciosl** [comment]
Yup, I'm afraid that this would break cache at public pages because of `class Admin::VisualizationsController < Admin::AdminController`, so this is a :-1: for the moment :(

Maybe this is a good point to invest in a small refactor of the public visualizations controller class hierarchy, I don't know if there's any reason for it to be under Admin (but maybe the effort is big).

**Comment by gfiorav** [comment]
We could also avoid this cache headers for public pages and embed maps so they would work as normal. What do you think?

**Comment by juanignaciosl** [comment]
IMHO exceptions at filters (`skip_filter`) should be used in exceptional cases, public pages are not. I _think_ there's something wrong in that hierarchy, why are public pages under Admin? I'd take a look and, if there's a good reason for them to be under admin, or refactor effort is huge or risky, I'd do what you say.

**Comment by javisantana** [comment]
I'd wait to think in refactors right now until we have a better idea of what is going to be the next editor version

**Comment by zenitraM** [comment]
I forgot that apart from Varnish there's the fact that the public pages and embeds won't be cached browser-side when this happens, so it's probably not a good idea

**Comment by gfiorav** [comment]
@javisantana Ok, but the problem of sensible info being displayed in some pages and being able to access it through browser history is going to remain the same no? Maybe we could use a separate controller for public stuff and reroute.

@zenitraM yeah! We can just avoid the change for public stuff since embeds and public pages don't contain sensible info anyway.

**Comment by juanignaciosl** [comment]
Ok, if javisantana suggest we should not refactor, let's keep this approach and skip the new headers for public pages :runner:

**Comment by gfiorav** [comment]
I think we should be in the clear with the skip filter applied. I'm deploying this to ded05 if anyone wants to do some testing of their own.

**Comment by gfiorav** [comment]
@juanignaciosl @zenitraM @javisantana 

I've done some testing. 

There is already some custom cache control so I'm bounding this to profile, account and api keys pages only. The rest remains untouched. I'm deploying this new version to staging to do some final testing.

I assume it's not too costly that account/profile pages are not cached in browser right?

**Comment by juanignaciosl** [comment]
Not costly and those are the most critical from a security point of view. You're also removing cache for billing page, isn't it?

**Comment by gfiorav** [comment]
[after out-of-github chat]

Billing is in central. We might need to do that there as well.

Is this :+1: ?

**Comment by juanignaciosl** [comment]
Yes, I think so :+1: I didn't know that page was now outside CartoDB itself (reported).

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -13,13 +13,13 @@ class SessionsController < ApplicationController
   ssl_required :new, :create, :destroy, :show, :unauthenticated, :account_token_authentication_error,
                :ldap_user_not_at_cartodb
 
+  skip_before_filter :ensure_org_url_if_org_user # Don't force org urls
```
Comment (juanignaciosl) [inline app/controllers/sessions_controller.rb]: Why is this reordering needed?

Comment (gfiorav) [inline app/controllers/sessions_controller.rb]: I always write skip_before_filter before to make sure it's not shadowing any bad code (i.e: one skip_before_filter is undoing another before_filter) 

Might be a wrong thing to do. Should I revert it?

Comment (juanignaciosl) [inline app/controllers/sessions_controller.rb]: Nice convention, just wanted to know the reason :+1: :-)

**Comment by gfiorav** [comment]
Ok, @juanignaciosl , if you give me a :+1: I'm deploying this tomorrow. Thanks!

**Pull request merged**